packages/strapi-generate-new/lib/before.js

```diff
@@ -45,8 +48,246 @@ module.exports = (scope, cb) => {
+                if (_.get(scope.database, 'port')) {
+                  return scope.database.port;
+                }
+
+                const ports = {
+                  mongo: 27017,
+                  postgres: 5432,
+                  mysql: 3306,
+                  sqlite3: 1433,
+                  redis: 6379
+                };
+
+                return ports[scope.client.database];
+              }
+            },
+            {
+              type: 'input',
+              prefix: '',
+              name: 'username',
+              message: 'Username:',
+              default: _.get(scope.database, 'username', undefined)
+            },
+            {
+              type: 'input',
+              prefix: '',
+              name: 'password',
+              message: 'Password:',
+              default: _.get(scope.database, 'password', undefined)
+            }
+          ])
+          .then(answers => {
+            scope.database.host = answers.host;
+            scope.database.port = answers.port;
+            scope.database.database = answers.name;
+            scope.database.username = answers.username;
+            scope.database.password = answers.password;
+
+            resolve();
+          });
+        }),
+        new Promise(resolve => {
+          let cmd = `npm install --prefix ${scope.rootPath}_ ${scope.client.connector}@alpha`;
+          if (scope.client.module) {
+            cmd += ` ${scope.client.module}`;
+          }
+
+          exec(cmd, () => {
+            if (scope.client.module) {
+              const lock = require(`${scope.rootPath}_/node_modules/${scope.client.module}/package.json`);
+              scope.client.version = lock.version;
+            }
+
+            resolve();
+          });
+        })
+      ];
+
+      Promise.all(asyncFn)
+      .then(() => {
+        if (scope.client.connector === 'strapi-bookshelf') {
+          const knex  = require(`${scope.rootPath}_/node_modules/knex`)({
+            client: scope.client.module,
+            connection: scope.database
+          });
+
+          knex.raw('select 1+1 as result').then(() => {
+            logger.info('Database connection is a success!');
+            knex.destroy();
+            execSync(`rm -r ${scope.rootPath}_`);
+
+            logger.info('Copying the dashboard...');
+
+            // Trigger callback with no error to proceed.
+            cb.success();
+          })
+          .catch(() => {
+            logger.warn('Database connection failed!');
+            connectionValidation();
+          });
+        } else if (scope.client.connector === 'strapi-mongoose') {
+          const Mongoose = require(`${scope.rootPath}_/node_modules/mongoose`);
+
+          Mongoose.connect(`mongodb://${ (scope.database.username && scope.database.password) ? `${scope.database.username}:${scope.database.password}@` : '' }${scope.database.host}:${scope.database.port}/${scope.database.database}`, {
+            useMongoClient: true
```

Comment:
This parameter is deprecated with the latest version of Mongoose. We are now using the 5.0.0-rc1.